openslides_backend/action/motion_category/number_motions.py

```diff
@@ -0,0 +1,203 @@
+                [
+                    "motion_ids",
+                    "motion_category_ids",
+                    "motions_number_with_blank",
+                    "motions_number_min_digits",
+                    "motions_amendments_prefix",
+                ],
+            )
+            self.meeting = meeting
+        else:
+            raise ActionException("Main category doesnt include meeting_id.")
+
+        gmr_categories = GetManyRequest(
+            Collection("motion_category"),
+            meeting.get("motion_category_ids", []),
+            ["prefix", "parent_id", "child_ids", "weight", "motion_ids"],
+        )
+        gmr_motions = GetManyRequest(
+            Collection("motion"),
+            meeting.get("motion_ids", []),
+            [
+                "lead_motion_id",
+                "category_weight",
+                "category_id",
+                "number",
+            ],
+        )
+        result = self.datastore.get_many([gmr_categories, gmr_motions])
+        self.mem_categories = result.get(Collection("motion_category"), {})
+        self.mem_motions = result.get(Collection("motion"), {})
+        self.mem_meetings = {category.get("meeting_id"): meeting}
+
+    def get_prefix(self, category_id: int) -> str:
+        """Get the prefix of a category. If none, get the prefix of the parent if exists."""
+        category = self.mem_categories.get(category_id, {})
+        if category.get("prefix"):
+            return category["prefix"]
+        elif (not category.get("parent_id")) or category_id == self.main_category_id:
+            return ""
+        else:
+            return self.get_prefix(category["parent_id"])
+
+    def get_lead_motion_id(self, motion_id: int) -> Optional[int]:
+        """Helper to get the lead_motions_id."""
+        motion = self.mem_motions.get(motion_id, {})
+        return motion.get("lead_motion_id")
+
+    def get_affected_categories(self, category_id: int) -> List[int]:
+        """Get all affected categories, inits and calls the helper."""
+        affected_categories: List[int] = []
+        queue = [category_id]
+        self.helper_affected_categories(queue, affected_categories)
+        return affected_categories
+
+    def helper_affected_categories(
+        self, queue: List[int], aff_categories: List[int]
+    ) -> None:
+        """Fill the aff_categories, tree walk by level."""
+        if not queue:
+            return
+        aff_categories.append(queue[0])
+        category = self.mem_categories.get(queue[0], {})
+        queue.extend(self.sort_category_children(category.get("child_ids", [])))
+        self.helper_affected_categories(queue[1:], aff_categories)
+
+    def sort_category_children(self, child_ids: List[int]) -> List[int]:
+        """Sort the categories by weight. Important for the helper_affected_categories."""
+        weighted_child_ids = []
+        for id_ in child_ids:
```

Comment:
Please rename all occurrences of `id_` and `ids_` to `id` and `ids`.

Reply:
id is a keyword, anyway renamed in id and ids.

Reply:
Now I see why everyone's been doing that lately... :D I also just had the same problem where I renamed a variable from `id` to something else and suddenly I had a string with `<built-in function>` in it. Maybe it's actually better to use `id_`

Anyway, you can leave it now, just a note.